src/ifs/wl_seat/touch_owner.rs

```diff
@@ -0,0 +1,197 @@
+use {
+    crate::{
+        fixed::Fixed,
+        ifs::wl_seat::WlSeatGlobal,
+        rect::Rect,
+        tree::{FindTreeUsecase, FoundNode, Node},
+        utils::clonecell::CloneCell,
+    },
+    ahash::AHashSet,
+    std::{cell::RefCell, rc::Rc},
+};
+
+pub struct TouchOwnerHolder {
+    default: Rc<DefaultTouchOwner>,
+    owner: CloneCell<Rc<dyn TouchOwner>>,
+}
+
+impl Default for TouchOwnerHolder {
+    fn default() -> Self {
+        Self {
+            default: Rc::new(DefaultTouchOwner),
+            owner: CloneCell::new(Rc::new(DefaultTouchOwner)),
+        }
+    }
+}
+
+impl TouchOwnerHolder {
```

Comment:
This also needs to handle the case where the node that we're interacting with changes its position. See `WlSeatGlobal::apply_changes`.

Reply:
What changes should be applied in such a case? There's no touch cursor, and for the duration of the touch sequence the node should stay the same no matter where it is.

Reply:
I think we'll have to store the current absolute positions of the touch ids in compositor space and then send synthetic motion events to the node when the node moves.